src/input/cursor.c

```diff
@@ -908,13 +910,22 @@ cursor_button_press(struct seat *seat, struct wlr_pointer_button_event *event)
 		return;
 	}
 
-	/* Handle _press_ on a layer surface */
+	/*
+	 * On press, set focus to a non-view surface that wants it.
+	 * Action processing does not run for these surfaces and thus
+	 * the Focus action (used for normal views) does not work.
+	 */
 	if (ctx.type == LAB_SSD_LAYER_SURFACE) {
 		struct wlr_layer_surface_v1 *layer =
 			wlr_layer_surface_v1_from_wlr_surface(ctx.surface);
 		if (layer && layer->current.keyboard_interactive) {
 			seat_set_focus_layer(seat, layer);
 		}
+#ifdef HAVE_XWAYLAND
+	} else if (ctx.type == LAB_SSD_UNMANAGED) {
+		desktop_focus_view_or_surface(seat, NULL, ctx.surface,
+			/*raise*/ false);
```

Comment:
Should we raise here as well?

Reply:
I'm not sure. Unmanaged surfaces are already on top of everything else (except possibly other unmanaged surfaces). Maybe it makes sense to restack the clicked one on top.

desktop_focus_view_or_surface() doesn't currently have code to raise an unmanaged surface, but it could be done.

Reply:
> Unmanaged surfaces are already on top of everything else (except possibly other unmanaged surfaces).

Only on the labwc scene graph side of things though, I am not sure if X11 internally has a unmanaged-always-on- top stacking order.

As we are already in the in the X11 ifdef + unmanaged check, maybe we could just restack it for the X11 side then do a `wlr_scene_node_raise_to_top()` for the scene graph side?

Reply:
I'd vote for _not_ adding any `wlr_xwayland_surface_restack()` calls for now. I'm saying that mostly out of fear of messing with OR surfaces, rather than having done any rational analysis (although I did check `sway` + `wlroots` and they only call the restack on map).

Reply:
> I am not sure if X11 internally has a unmanaged-always-on- top stacking order.

We actually restack unmanaged surfaces on top right after raising another (non-OR) XWayland surface. At that moment the labwc and XWayland stacking order should be in sync. Not sure if/how long it stays that way.

I'm good with leaving the current stacking approach as-is unless/until we run into specific issues with it.

Reply:
> I'm good with leaving the current stacking approach as-is unless/until we run into specific issues with it.

Me as well.